commands/command_migrate_info.go

```diff
@@ -164,15 +164,24 @@ func migrateInfoCommand(cmd *cobra.Command, args []string) {
 		},
 
 		TreePreCallbackFn: func(path string, t *gitobj.Tree) error {
-			if migrateFixup && path == "/" {
-				var err error
+			if migrateFixup {
+				if path == "/" {
+					var err error
 
-				fixups, err = gitattr.New(db, t)
-				if err != nil {
-					return err
+					fixups, err = gitattr.New(db, t)
+					if err != nil {
+						return err
+					}
 				}
 				return nil
 			}
+
+			for _, e := range t.Entries {
+				if e.Name == ".gitattributes" && e.IsLink() {
+					return errors.Errorf("migrate: %s", tr.Tr.Get("expected '.gitattributes' to be a file, got a symbolic link"))
+				}
```

Comment:
It's likely that `.gitattributes` will be early on in the tree.  If we've seen it, can we break out of the loop even if it's not a link?  That would help us perform better on repos with large trees.

Reply:
Oh, good point; I should have pulled over that same logic from the `import` command's version.